File: src/app/task.ts
import { TaskState } from './task-state.enum'
export class Task {
    description: string = "Empty";
    state: TaskState = TaskState.TODO;
    dueDate: Date = new Date();
    constructor(description: string, state: TaskState, dueDate: Date) {
        this.description = description;
        this.state = state;
        this.dueDate = dueDate;
    }
}

File: src/main/java/com/coyoapp/tinytask/domain/Task.java
package com.coyoapp.tinytask.domain;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.sql.Date;
import java.time.Instant;

@Table(name = "task")
@Entity
@Setter
@Getter
@EntityListeners(AuditingEntityListener.class)
public class Task {

  @Id
  @GeneratedValue(generator = "uuid2")
  @GenericGenerator(name = "uuid2", strategy = "uuid2")
  @Column(name = "id", nullable = false, updatable = false)
  private String id;

  private String name;

  @CreatedDate
  private Instant created;

  @Enumerated(EnumType.STRING)
  private State state;

  @ManyToOne
  @JoinColumn(name = "user_id", nullable = false)
  private User user;

  private Date dueDate;

}
